project_wip_material/task_material.py

```diff
@@ -349,6 +355,31 @@ def unlink(self):
         return super().unlink()
 
 
+class TaskMaterialWithProjectSelect(models.Model):
+    """Add a technical computed field to allow constrain the domain when selecting the task.
+
+    This field is not stored and only used as a helper in the global list view of material.
+    """
+
+    _inherit = 'project.task.material'
+
+    project_select_id = fields.Many2one(
+        'project.project',
+        'Project Select',
+        compute='_compute_project_select',
+        inverse=lambda self: None,
+    )
+
+    def _compute_project_select(self):
+        for material in self:
+            material.project_select_id = material.project_id
+
+    @api.onchange('project_select_id')
+    def _onchange_project_select_reset_task(self):
+        if self.project_select_id != self.task_id.project_id:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Access to member 'task_id' before its definition line 380](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4181253)